frappe/public/js/frappe/model/workflow.js

```diff
@@ -6,11 +6,19 @@ frappe.provide("frappe.workflow");
 frappe.workflow = {
 	state_fields: {},
 	workflows: {},
+	avoid_status_override: {},
 	setup: function (doctype) {
 		var wf = frappe.get_list("Workflow", { document_type: doctype });
 		if (wf.length) {
 			frappe.workflow.workflows[doctype] = wf[0];
 			frappe.workflow.state_fields[doctype] = wf[0].workflow_state_field;
+			frappe.workflow.avoid_status_override[doctype] = frappe.workflow.workflows[
+				doctype
+			].states
+				.filter((row) => {
+					if (row.avoid_status_override) return true;
+				})
+				.map((d) => d.state);
```

Comment:
```suggestion
			frappe.workflow.avoid_status_override[doctype] = wf[0].states
				.filter((row) => row.avoid_status_override)
				.map((d) => d.state);
```

Reply:
I was planning to make the review change in the evening, sorry :grimacing: